client.go

```diff
@@ -115,14 +130,18 @@ func (c *Client) handleConn(ctx context.Context, conn net.Conn) error { //nolint
 					break
 				}
 
-				c.log.Warn("Message lost", lctx.Uint64("id", exp.probe.ID), lctx.Str("data", exp.probe.Data))
+				c.log.Warn("Data dropped",
+					lctx.Uint64("expected_id", resp.probe.ID),
+					lctx.Uint64("id", exp.probe.ID),
+					lctx.Str("data", exp.probe.Data),
+				)
```

Comment:
Be careful, with UDP packets might come back in the wrong order, and here since we always check the first message only to see if it's the one we expect, we might think that it's missing when instead we just received the next one before the previous one.

For example it could happen that we send 1 packet per second, and for 5 seconds we get nothing, then we suddenly receive all 5 responses back in random order, together. This should be noted as shuffled responses, but not considered dropped packets.

Reply:
From what was suggested, I assumed the packet sequence would be handled more gracefully in a different iteration/PR.

> As a second iteration, we could send larger packets, with a more complex protocol where we send the number of packets as part of the header. The server could just send back what it received straight away without much added complexity, but it would be up to the client to do the chunking, unchunking, and check for missing or shuffled packets based on packet sequence.

But I agree that an accurate log message is key. I added a commit where this is re-phrased, please check again. If you still prefer a different _behavior_, let me know.

I wanted to avoid implementing a re-order for now mainly because I was afraid of a memory leak with an ever-growing slice when the expected messages never get received. So we would need to expire messages, which overall seemed out of scope. Maybe I am thinking too complicated on this.